variables.tf

```diff
@@ -86,3 +86,35 @@ variable "cbr_rules" {
   default     = []
   # Validation happens in the rule module
 }
+
+##############################################################################
+# Event Notification
+##############################################################################
+
+variable "skip_en_iam_authorization_policy" {
+  type        = bool
+  description = "Set to true to skip the creation of an IAM authorization policy that permits all Secrets Manager instances (scoped to the resource group) an 'Event Source Manager' role to the given Event Notifications instance passed in the `existing_en_instance_crn` input variable. In addition, no policy is created if `enable_event_notification` is set to false."
+  default     = false
+}
+
+variable "enable_event_notification" {
+  type        = bool
+  default     = false
+  description = "Set this to true to enable lifecycle notifications for your Secrets Manager instance by connecting an Event Notifications service. When setting this to true, a value must be passed for `existing_en_instance_crn` variable."
+}
+
+variable "existing_en_instance_crn" {
+  type        = string
+  default     = null
+  description = "The CRN of the Event Notifications service to enable lifecycle notifications for your Secrets Manager instance."
+}
+
+variable "endpoint_type" {
+  type        = string
+  description = "The endpoint type of the Secrets Manager instance required for establishing a connection between an Event Notifications service. Possible values are `public`, `private`."
```

Comment:
Its not the secrets manager instance endpoint right? Its the event notification service endpoint. So you are telling the secrets manager instance which endpoint to use when communicating with the the event notification service

Reply:
No, it is to tell the event notification which endpoint of the secret manager to use for registration and communication.
For example if create a SM with `service_endpoint` as `public-and-private`. The SM will have both public and private endpoints to choose from and if we choose `private` for `endpoint_type` during registration it will use the private endpoint of the SM to register and communicate.
![Screenshot 2024-01-11 at 8 49 32 AM](https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/assets/122446118/6503b68e-69e4-45c2-ab9b-5d5924b18a22)

Screenshot of a failure, when i tried to set `endpoint_type` as `private`. It was not able to register the SM.

Reply:
oh ok - well if this is the case, then we should add some validation logic to ensure that `endpoint_type` cannot be `public` if `service_endpoints` is set to `private`